api/gateway-operator/v2alpha1/controlplane_types.go

```diff
@@ -352,6 +357,28 @@ type ControlPlaneConfigDump struct {
 	DumpSensitive ConfigDumpState `json:"dumpSensitive"`
 }
 
+// ControlPlaneObjectFilters defines filters to limit watched objects by the controllers.
+type ControlPlaneObjectFilters struct {
+	// Secrets defines the filters for watched secrets.
+	//
+	// +optional
+	Secrets *ControlPlaneFilterForObjectType `json:"secrets,omitempty"`
+	// ConfigMaps defines the filters for watched config maps.
+	//
+	// +optional
+	ConfigMaps *ControlPlaneFilterForObjectType `json:"configMaps,omitempty"`
+}
+
+// ControlPlaneFilterForObjectType defines the filters for a certain type of object.
+type ControlPlaneFilterForObjectType struct {
+	// MustHaveLabel specifies a label that an object must have the label with value "true" to get watched by the controllers.
+	// For example, if the secrets.mustHaveLabel is set to "some-label", only secrets with label "some-label=true" are reconciled.
+	//
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	MustHaveLabel *string `json:"mustHaveLabel,omitempty"` // REVIEW: add other validation of label keys, like the prefix/name format, max length, used chars?
```

Comment:
> // REVIEW: add other validation of label keys, like the prefix/name format, max length, used chars?

I think we'll want to add that when we get product / user demand for this.

Perhaps a good option for making the API extensible would be to make `ControlPlaneFilterForObjectType` a union type, i.e. add a `filterType` (or better named) field which would indicate which filtering option is chosen. Initially only the mustHabeLabel would be supported but it would allow us to extend this API in the future.

```
configMaps:
  filterType: mustHaveLabel
  mustHaveLabel:
    name: my-label
```

WDYT?